### PR TITLE
[xcode11] Fix build and bump versions for Xcode 11.0 beta 1

### PR DESCRIPTION
**Pull request by mandel-macaque**
Allows to build `xamarin-macios` using **Xcode 11**.
In order to do so, until we get the mono SDK built in the bots with Xcode 11, you need to set the environment variable `MONO_BUILD_FROM_SOURCE=1`.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Won't work yet until we do merge Xcode11 with master

**Comment by mandel-macaque** [comment]
Current failure when trying to build mono: https://gist.github.com/mandel-macaque/9a29f82df199a16c946fd9fb446336a4

**Comment by mandel-macaque** [comment]
We have created a bug to fix the workaround needed to compile: https://github.com/mono/mono/issues/14792

File: system-dependencies.sh [hunk 1]
```diff
@@ -78,7 +78,7 @@ while ! test -z $1; do
 			unset IGNORE_HOMEBREW
 			PROVISION_SHARPIE=1
 			unset IGNORE_SHARPIE
-			PROVISION_SIMULATORS=1
+			#PROVISION_SIMULATORS=1
```
Comment (chamons) [inline system-dependencies.sh]: Can we file an issue so we remember to turn this back on?

Comment (mandel-macaque) [reply system-dependencies.sh]: done: https://github.com/xamarin/xamarin-macios/issues/6205

File: .gitmodules [hunk 2]
```diff
@@ -8,7 +8,7 @@
 [submodule "external/mono"]
     path = external/mono
     url = ../../mono/mono.git
-    branch = 2019-02
+    branch = 2019-02-xcode11
```
Comment (spouliot) [inline .gitmodules]: Are we sure that the changes are not safe / compatible with current stable Xcode ?
Ideally we would not have to periodically update (merge) mono... too easy to forget

**Comment by mandel-macaque** [comment]
Everything compiles at this point, but tests fail to build. We can do that in this branch or in a diff one.

File: .gitmodules (hunk 2 above)
Comment (mandel-macaque) [reply .gitmodules]: @spouliot we have a workaround, but I have filled and issue https://github.com/mono/mono/issues/14792 it has already been assigned by @marek-safar (I don't have the super powers to assign it to myself, but is an easy issue to fix).

**Comment by spouliot** [comment]
Xtro failure requires a new sharpie @dalexsoto and https://github.com/xamarin/xamarin-macios/pull/6207 (can be landed before) and then it will report correctly (i.e. more things to fix)

**Comment by rolfbjarne** [comment]
`do-not-merge` until we've fixed the mlaunch/CoreSimulator issue on other branches.

File: .gitmodules (hunk 2 above)
Comment (akoeplinger) [reply .gitmodules]: We have the workaround in 2019-02 now and a PR open for the casting fix, so you should be able to switch to 2019-02 again.

**Comment by rolfbjarne** [comment]
> * merge backport PRs that don't require Xcode11

Those PRs will fail because the xcode11 branch doesn't have the mlaunch fix.

**Comment by spouliot** [comment]
@rolfbjarne yes, but we can bump mlaunch without bumping to Xcode 11 right ? that means more green - both before xcode11 bump and after

**Comment by spouliot** [comment]
in fact we need to merge `d16-2` periodically into `xcode11` and that fix is already there... doing it now solves problems :)

**Comment by rolfbjarne** [comment]
#6216 has now been fixed in all relevant branches (master, d16-2, d16-1), so removing the `do-not-merge` label.

**Pull request merged**